code-challenges.js

```diff
@@ -16,41 +22,137 @@ const fruit2 = "banana"
+    //Back to the pseudo coding drawing board.
+// Attempt #2 in the form of a function expression.
+    // Declare a function with the parameter of split to pass currentCohort through.
+    // Use the return with the parameter of string.split("") to turn the currentCohort string into an array of letters.
+    // Create a variable equal to the output of function
+    // Console.log the new variable to get the output
+    // Create a variable equal to the previous variable .reverse
+    // Console.log this new variable with the .reverse
+    // Then, I was going to use the .join method to put it all back together, but this isn't working.
+    // const split = (string) => {
+    //     return string.split("")
+    //  }
+    // console.log(split(currentCohort)) // Output ["D", "e", "l", "t", "a", " ", "2", "0", "2", "2"]
+    // // Now we have an array of elements instead of a string
+    
+    // var array = console.log(split(currentCohort))
+    // console.log(array) // // Output ["D", "e", "l", "t", "a", " ", "2", "0", "2", "2"]
+    // var reverseArray = array.reverse()
+    // console.log(reverseArray) // TypeError: Cannot read properties of undefined (reading 'reversed')
+//Attempt #3
+// Research new approaches to the problem
+// Between the syllabus notes and some other reseaarch, I am finding I need to write a function using the keyword function. Can you explain more about this syntax and the difference between a function expression and function declaration? Is the keyword function being used to define an expression here? Just trying to grasp the theory.
+// Write a function, then name, then parameter, and curly brackets.
+// Set parameter to (string)
+// Return the parameter with chained methods starting with the .split("") to create an array, then the .reverse method to reverse the elements of an array, and finally the .join method to join the reversed elements of the array back together.
+// Lastly console.log the name of the function expression with the parameter of currentCohort.
+// 
 const currentCohort = "Delta 2022"
 
+function reverseString(string) {
+    return string.split("").reverse().join("")
+}
+console.log(reverseString(currentCohort))
 
 // --------------------4) Create a statement that will evaluate whether each number in an array is even or odd. Use the test variable provided below. Expected output: "odd" "even" "odd" "even" "odd" "even"
 
 // Pseudo code:
+// Create a for loop to iterate through the array to evaluate whether each element is odd or even.
+// After let, state where the index starts, then we iterate through the length of the array, and what to do after each loop (increments).
+// Filter the array using if else statements to check if conditions are met and console.log those results
+// Final step, console.log(myArray) for output.
+
+ const myArray = [13, 34, 5, 10, 27, 42]
 
-const myArray = [13, 34, 5, 10, 27, 42]
+for(i=0; i<myArray.length; i++) {
```

Comment:
You mentioned wanting to understand this line a little better.  i is a variable that is set to in a sense, be our index (which is why we are initializing it to 0 since arrays are zero indexed).  The next section is saying when we want the iteration to stop (otherwise it will go on forever).  We do this by determining what the length is of the array and saying as long as the current index or i is less than our arrays length, we will continue to move to the next value in my array.  i++ is to advance to the next iteration, in other words, once the code block inside the for loop (such as a conditional statement) has been completed, we need to move forward to the next element which is where i++ comes in.